pkg/flare/archive_security.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package flare
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mholt/archiver"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/status"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// CreateSecurityAgentArchive packages up the files
+func CreateSecurityAgentArchive(local bool, logFilePath string) (string, error) {
+	zipFilePath := getArchivePath()
+	return createSecurityAgentArchive(zipFilePath, local, logFilePath)
+}
+
+func createSecurityAgentArchive(zipFilePath string, local bool, logFilePath string) (string, error) {
```

Comment:
There is a lot of code duplication with `pkg/flare/archive.go:createArchive`. Do we need a separate function ? Why not generate a generic flare ?
If this is needed we should rework both function to avoid code duplication. We could pass a list of `zipXXX` function to call ?

Reply:
Good point, thank you, let me see which option works best here.

Reply:
It's actually closer to `createDCAArchive`, the primary differences are how we generate the status and zipping the compliance benchmark files. Let me extract some helpers, I am hesitant to change the node agent's version too much due to lack of tests in this area.